### PR TITLE
Fixes for applying tasty-mima across Scala 2 and Scala 3

### PR DESCRIPTION
**Pull request by sjrd**
@nicolasstucki Do you want to review this one? There's not everything yet, notably the `InternalError`s, but it's already something.

### CODE REVIEW

File: tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 1]
```diff
@@ -692,6 +736,8 @@ private[pickles] class PickleReader {
 
 private[reader] object PickleReader {
 
+  private val MangledDefaultGetterNameRegex = "^(.+)\\$default\\$([0-9]+)$".r
```
Comment (nicolasstucki) [inline tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:739]: Does this work with default constructor arguments?  Maybe add a test for those.

Comment (nicolasstucki) [inline tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:739]: Does this work with default constructor arguments?  Maybe add a test for those.

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

File: tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala (hunk 1 above)
Comment (sjrd) [reply tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:739]: It works. I added a test :)

**Pull request merged**